component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordResetEnforcer.java

```diff
@@ -400,4 +398,21 @@ private void updateAuthenticatedUserInStepConfig(AuthenticationContext context,
         }
         context.setSubject(authenticatedUser);
     }
+
+    private String getLastPasswordUpdateTime(UserStoreManager userStoreManager, String tenantAwareUsername)
+            throws org.wso2.carbon.user.core.UserStoreException {
+
+        Claim[] claims = userStoreManager.getUserClaimValues(tenantAwareUsername, null);
+        if (claims != null) {
+            for (Claim claim : claims) {
+                if (PasswordPolicyConstants.LAST_CREDENTIAL_UPDATE_TIMESTAMP_CLAIM.equals(claim.getClaimUri()) ||
```

Comment:
Can we guarantee that the order of claims? What if the non-identity claim comes first in the array for the migrated users?